### PR TITLE
feat(matchmaker): RPC API stub

### PR DESCRIPTION
**Pull request by szabba**

### CODE REVIEW

File: matchmaker/rpc/v1/api.proto [hunk 1]
```diff
@@ -0,0 +1,37 @@
+syntax = "proto3";
```
Comment (szabba) [inline matchmaker/rpc/v1/api.proto:1]: @phaux If we followed this route, this file + the README I plan to add would be the most interesting to you.

**Comment by phaux** [comment]
Add README with instructions how to run and example `curl` requests and responses

File: matchmaker/README.markdown [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+You have several options.
+
+1. Geneerate a Connect RPC client from the API spec.
+   So far this is only possible for Go.
+2. Generate a gRPC / gRPC-Web client from the API spec.
+   `connect-go` is compatible with those.
+3. Use cURL / an HTTP client.
+   - See the docs on [invoking Connect APIs with HTTP clients].
+   - The [JSON <-> Protobuf mapping] is documented separately
```
Comment (phaux) [inline matchmaker/README.markdown:15]: Can we generate HTTP API description and include it here or a separate md file?

File: matchmaker/README.markdown [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+[invoking Connect APIs with HTTP clients]: https://connect.build/docs/browsers-and-curl
+[JSON <-> Protobuf mapping]: https://developers.google.com/protocol-buffers/docs/proto3#json
+
+## File structure
```
Comment (phaux) [inline matchmaker/README.markdown:20]: Readme should be more concise and this section isn't needed at all IMO

File: matchmaker/README.markdown [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+## Configuration
+
+The following environment variables are supported.
+
+ **Name** | **Description** | **Example**
+---|---|---
+`LOG_LEVEL` | A zerolog-compatible log level value. | `info` (default)
+`PORT` | The port for the HTTP server to listen at. | `4447` (default)
+`ZONEINFO` | Preferred [tz database] location. See [`time.LoadLocation`] | `/zoneinfo` (default: empty)
```
Comment (phaux) [inline matchmaker/README.markdown:43]: Server location shouldn't matter

**Review by phaux — CHANGES_REQUESTED** [review]

**Comment by phaux** [comment]
I think the readme structure should be:

- Usage
    1. List of required tools and their versions
    2. How to build 
    3. List of config files, env vars, cmd line args
    4. How to run 
    5. Description of exposed API
- Development
    - List of extra required tools if any
    - How to run in watch mode if applicable
    - What to run before commit (fmt, lint, generate)
    - How to deploy (in the future)
 
Things that will turn out to be the same for all packages in monorepo will be factored out to the main README later.

File: matchmaker/buf.yaml [hunk 5]
```diff
@@ -0,0 +1,7 @@
+version: v1
+breaking:
+  use:
+    - FILE
+lint:
+  use:
+    - DEFAULT
```
Comment (szabba) [inline matchmaker/buf.yaml:7]: Add linting rule that only allows unary RPCs - streaming is a whole bunch of problems we don't need to get into.

**Comment by szabba** [comment]
Make sure to regenerate the code in `matchmaker/gen/` before merging.

File: matchmaker/README.markdown (hunk 2 above)
Comment (szabba) [reply matchmaker/README.markdown:15]: There's no preexisting tool that just does that. We could write one, but that's a siginificant effort and OT for this PR/repository.

File: matchmaker/README.markdown (hunk 4 above)
Comment (szabba) [reply matchmaker/README.markdown:43]: (See also #3 for doing the same with the trusted CA root certificate pool.)

**Comment by szabba** [comment]
> Make sure to regenerate the code in `matchmaker/gen/` before merging.

I've added a basic pipeline workkflow, since the Earthly build verifies it automatically already.

File: matchmaker/README.markdown (hunk 2 above)
Comment (phaux) [reply matchmaker/README.markdown:15]: Then write one by hand

File: matchmaker/README.markdown (hunk 3 above)
Comment (szabba) [reply matchmaker/README.markdown:20]: I've reorganized the readme roughly along the lines of the other, longer comment. How is it now?

File: matchmaker/README.markdown (hunk 2 above)
Comment (szabba) [reply matchmaker/README.markdown:15]: Let's talk about incentives here:
- OpenAPI is relatively general and unopinionated in what it describes, but is (I feel): 
    - very verbose, 
    - makes people care about _mostly_ uninteresting details.
- Manually written examples _will_ go out of sync. It's just a matter of time.
- A tool for generating swagger out of the proto spec is not readily available.
- I think I've found less effort-intensive way to implement generating the curl samples with [`http2curl`](https://pkg.go.dev/github.com/moul/http2curl) but I'd prefer to deliver that in a separate PR.

**Review by phaux — APPROVED** [review]

File: matchmaker/buf.yaml (hunk 5 above)
Comment (szabba) [reply matchmaker/buf.yaml:7]: Done.

**Pull request merged**